From: hololeap <hololeap@protonmail.com>
Signed-off-by: hololeap <hololeap@protonmail.com>

Add `executable` flag to toggle building the `fourmolu` executable.

This executable is not normally needed when building fourmolu as a dependency
for other packages.

diff -urN fourmolu-0.4.0.0/fourmolu.cabal fourmolu-0.4.0.0-r1/fourmolu.cabal
--- fourmolu-0.4.0.0/fourmolu.cabal	2022-01-22 15:31:59.668891373 -0700
+++ fourmolu-0.4.0.0-r1/fourmolu.cabal	2022-01-22 15:32:07.018891376 -0700
@@ -114,6 +114,10 @@
     if impl(ghc <8.10.0)
         ghc-options: -fmax-pmcheck-iterations=3000000
 
+flag executable
+    description: Build the fourmolu executable
+    default: False
+
 executable fourmolu
     main-is:          Main.hs
     hs-source-dirs:   app
@@ -139,6 +143,9 @@
     else
         ghc-options: -O2 -Wall -rtsopts
 
+    if !flag(executable)
+        buildable: False
+
 test-suite tests
     type:               exitcode-stdio-1.0
     main-is:            Spec.hs
